clash-prelude/src/Clash/Explicit/Signal.hs

```diff
@@ -898,3 +900,5 @@ convertReset clkA clkB (unsafeToHighPolarity -> rstA0) =
       (SAsynchronous, SSynchronous) ->
         delay clkB enableGen True $
           delay clkB enableGen True rstA1
+
+{-# RULES "sequenceAVecSignal" Clash.Sized.Vector.traverse# id = vecBundle# #-}
```

Comment:
Is this actually better? `vecBundle#` is defined as `traverse# id`
Also, should this rule be defined in `Clash.Signal.Bundle`? Then it wouldn't be an orphan

Reply:
It's better for synthesis because of this magic rule: 

https://github.com/clash-lang/clash-compiler/blob/264e5a4a7e0f818bae986ac20f8ea2b23f37a888/clash-ghc/src-ghc/Clash/GHC/GHC2Core.hs#L447

https://github.com/clash-lang/clash-compiler/blob/264e5a4a7e0f818bae986ac20f8ea2b23f37a888/clash-ghc/src-ghc/Clash/GHC/GHC2Core.hs#L979-L1004

Also, we can't define it in `Clash.Signal.Bundle`, because then  it gets applied in said module, and we end up with `vecBundle# = vecBundle#` (found this out during implementation of this PR)